### DISCUSSION TITLE
YamlCreate.ps1

### DISCUSSION THREAD

**Discussion by Trenly** [microsoft/winget-pkgs#21005] [Ideas] [answered]
Is there a reason that the YamlCreate powershell script doesn't listen for keypresses and goes based on strings when selecting whether to create a new manifest or update an existing one? Would this be something that could be, or should be, changed about it?

**Comment by OfficialEsco** [comment, marked as answer]
If you know some Powershell, feel free to go crazy with improving the YamlCreate script, i would love more contributors...
If you just have ideas, i could try to add them, but since none is giving feedback or helping with the code i have no idea what to do 🤷‍♂️

It kinda also badly need a better way to process the yaml file instead of reading line by line in raw, someone suggested using [YamlDotNet](https://www.nuget.org/packages/YamlDotNet/) but i'm unsure what we can use..

The current YamlCreate.ps1 is basically a beefed up version of the old one which KevinLaMS created over a year ago [(Latest small fix)](https://github.com/chrisgalvan/winget-pkgs/blob/5d1c200120f1b18dcc92a0c686aea1344ccee990/Tools/YamlCreate.ps1)

Also [wingetcreate](https://github.com/microsoft/winget-create) have the potential to be a way better CLI Yaml Creator, however the development is.. slow.. and its missing the features from YamlCreate which i expected to be in v0.1

**Comment by denelon** [comment]
Feel free to make suggestions for the [Windows Package Manager Manifest Creator](https://github.com/microsoft/winget-create) as well.  We want to make authoring and validating manifests easier without compromising on quality. YamlCreate has been very helpful, and iterates quickly, so use what tool works for you.

**Reply by Trenly** [reply]
I will certainly look at filing a few over there. The last time I used the manifest creator was probably about two weeks ago, when I found that for my particular habits, using the powershell script was easier, especially given that it prompts for all the manifest fields
